### PR TITLE
feat(FX-4322): Auctions Watch Lot - In Product Information Overlay

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Feat**

This PR solves [FX-4322]

Review app: [watch-log-pop-up](https://watch-log-pop-up.artsy.net/)

Live auctions: [with specified registration deadline](https://watch-log-pop-up.artsy.net/auction/dima-live-auction-test-1), [without specified registration deadline](https://watch-log-pop-up.artsy.net/auction/dima-live-auction-test-2)

### Description
* We will only display the “register to bid” button (“registration pending”, “verify identity”, “approved to bid” and etc buttons **will not** be displayed) in the pop-up
* We show the pop-up until the user register (pressed “register to bid” button)
* Do not show popover if registration deadline is specified and has already passed

#### With specified registration deadline
https://watch-log-pop-up.artsy.net/artwork/634ed29627729d00066067aa

https://user-images.githubusercontent.com/3513494/196486036-cc814e9f-1da1-4761-9a2b-d44f46f6858b.mp4

#### Without specified registration deadline
https://watch-log-pop-up.artsy.net/artwork/634ed371bd5df100066e34b2

https://user-images.githubusercontent.com/3513494/196490382-013f416e-2517-45c8-8b46-5f3c5db2d591.mp4

[FX-4322]: https://artsyproduct.atlassian.net/browse/FX-4322?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by nickskalkin — DISMISSED** [review]
👍

File: src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkActionsSaveButton.tsx [hunk 1]
```diff
@@ -4,40 +4,86 @@ import {
...
-      <UtilButton
-        name="bell"
-        Icon={isSaved ? FilledIcon : BellIcon}
-        label="Watch lot"
-        onClick={handleSave}
-      />
+      <Popover
+        title={
+          <Text variant="sm-display">
+            Register ahead of time to bid in this auction and get notifications
```
Comment (dblandin) [inline src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkActionsSaveButton.tsx:58]: **suggestion**: We might consider adding this copy to our [translations file](https://github.com/artsy/force/blob/fab739545f31ae264d6c660aed252e0e15822188/src/System/i18n/locales/en-US/translation.json).

Comment (dblandin) [reply src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkActionsSaveButton.tsx:58]: I think we also discussed a slight copy change to remove "ahead of time" when the auction does not have a specific registration deadline.

Registration deadline: Register ahead of time to bid in this auction and get notifications for this lot.

No registration deadline: Register to bid in this auction and get notifications for this lot.

**Review by dblandin — COMMENTED** [review]
Looking good! I noticed one state that we might want to account for which is when you're unauthenticated, click Watch Lot, and sign up / login via the modal.

When I complete that flow, I'm watching the lot but I don't see the popover.

**Comment by dimatretyak** [comment]
> When I complete that flow, I'm watching the lot but I don't see the popover

I think it's better to add these changes in a separate PR. do you mind?

File: src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkActionsSaveButton.tsx (hunk 1 above)
Comment (dimatretyak) [reply src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkActionsSaveButton.tsx]: [Added changes](https://github.com/artsy/force/pull/11171/commits/ef77693296d52ac794136916097252eb1ce15472) and also updated review app

**Review by gkartalis — APPROVED** [review]

**Pull request merged**